File: 백준/Bronze/2884. 알람 시계/알람 시계.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String s = br.readLine();
        StringTokenizer st = new StringTokenizer(s, " ");
        int h = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());

        if(m >= 45)
            System.out.println(h + " " + (m-45));
        else {
            if(h-1 < 0)
                h = 23;
            else
                h -= 1;
            m = 60-(45-m);
            System.out.println(h + " " + m);
        }
    }
}

File: 백준/Bronze/2884. 알람 시계/알람 시계.kt
fun main() {
    val (h, m) = readLine()!!.split(" ").map { it.toInt() }
    if (m >= 45) {
        println("$h ${m - 45}")
    } else {
        val nh = if (h == 0) 23 else h - 1
        val nm = m + 60 - 45
        println("$nh $nm")
    }
}